CRM/Pledge/BAO/Pledge.php

```diff
@@ -1079,13 +1079,14 @@ public static function updatePledgeStatus($params) {
                   $activityType,
                   'name'
                 ),
-                'activity_date_time' => CRM_Utils_Date::isoToMysql($now),
+                'activity_date_time' => CRM_Utils_Date::isoToMysql(date('YmdHis')),
```

Comment:
since you are calling the api to create this it could be simplified to 

'activity_date_time' => 'now',

or simply omit the line since I am pretty sure that's the default for new activities via the api.